ext/standard/string.c

```diff
@@ -1233,12 +1233,12 @@ PHP_FUNCTION(implode)
 	ZEND_PARSE_PARAMETERS_START(1, 2)
 		Z_PARAM_ARRAY_HT_OR_STR(arg1_array, arg1_str)
 		Z_PARAM_OPTIONAL
-		Z_PARAM_ARRAY_HT_OR_NULL(pieces)
+		Z_PARAM_ARRAY_HT(pieces)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (pieces == NULL) {
 		if (arg1_array == NULL) {
-			zend_type_error("%s(): Argument #1 ($pieces) must be of type array, string given", get_active_function_name());
+			zend_type_error("%s(): Argument #1 ($array) must be of type array, string given", get_active_function_name());
```

Comment:
```suggestion
			zend_type_error("%s(): Argument #2 ($array) must be of type array, string given", get_active_function_name());
```
Isn't it argument #2?

Reply:
It's an argument number 2 only when it's not an argument number 1. In this case, it is an argument number 1.